test/libsolidity/SolidityNameAndTypeResolution.cpp

```diff
@@ -2044,6 +2044,93 @@ BOOST_AUTO_TEST_CASE(external_visibility)
 	CHECK_ERROR(sourceCode, DeclarationError, "Undeclared identifier.");
 }
 
+BOOST_AUTO_TEST_CASE(similar_name_suggestions_expected)
+{
+	char const* sourceCode = R"(
+		contract c {
+			function func() {}
+			function g() public { fun(); }
+		}
+	)";
+	CHECK_ERROR(sourceCode, DeclarationError, "Undeclared identifier. Did you mean \"func\"?");
```

Comment:
Ah, it's  so nice seeing this  :-)

Reply:
I feel this error message might be better?
Undeclared identifier "fun". Did you mean "func"?

Reply:
We should have the source location highlighting the actual culprit (the identifier), but I think it can always help in clarity.